main.tf

```
@@ -103,18 +106,48 @@ locals {
   }
 
   ## Create the node groups, one for each instance type AND each availability zone/subnet
-  node_groups = {
+  v2_node_groups = tomap({
     for node_group in flatten([
       for instance_type in var.node_pool_instance_types : [
         for i, j in data.aws_subnet.private_subnets : {
           subnet_ids     = [data.aws_subnet.private_subnets[i].id]
           instance_types = [instance_type]
           name           = "snc-${split(".", instance_type)[1]}-${data.aws_subnet.private_subnets[i].availability_zone}"
-          labels         = merge(var.node_pool_labels, { "cloud.streamnative.io/instance-type" = lookup(local.compute_units, split(".", instance_type)[1], "null") })
+          taints         = {}
+          desired_size   = var.node_pool_desired_size
+          min_size       = var.node_pool_min_size
+          max_size       = var.node_pool_max_size
+          labels         = tomap(merge(var.node_pool_labels, { "cloud.streamnative.io/instance-type" = lookup(local.compute_units, split(".", instance_type)[1], "null") }))
         }
       ]
     ]) : "${node_group.name}" => node_group
-  }
+  })
+
+  v3_node_taints = var.enable_v3_node_taints ? {
+    "core" = {
+      key    = "node.cloud.streamnative.io/core"
+      value  = "true"
+      effect = "NO_SCHEDULE"
+    }
+  } : {}
+
+  v3_node_groups = tomap({
+    "core" = {
```

Would be better to keep consistent with the `name`

```suggestion
    "snc-core" = {
```

---

i think updating this for consumers of the module after this commit will likely add in another node group and/or remove the one they have so we'll have to be careful applying this in the future to things already using it. but i'm fine with the change.